### PR TITLE
LPS-84043 Consistency: Apply to more cases

### PR DESCRIPTION
**Pull request by robertoDiaz**

### CODE REVIEW

**Comment by sergiogonzalez** [comment]
Hey @robertoDiaz @renatodepaiva
For this kind of changes we need to have some integration tests to validate that the lastPostDate is actually the expected. We need to cover the different cases we have updated in this source code. We already have some tests checking that behaviour, see https://github.com/liferay/liferay-portal/blob/master/modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLCheckInCheckOutTest.java#L203 as an example.
Please, add new integration tests to cover those use cases and resend.
Thanks

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java [hunk 1]
```diff
@@ -1158,6 +1173,13 @@ public DLFolder updateStatus(
 
 		dlFolderPersistence.update(dlFolder);
 
+		// Parent folder
+
+		if (status == WorkflowConstants.STATUS_APPROVED) {
```
Comment (sergiogonzalez) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java]: We need to check that `dlFolder.getParentFolderId()` is not 0 because otherwise this will fail?

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java [hunk 2]
```diff
@@ -257,6 +251,11 @@ public DLFolder deleteFolder(
 
 		deleteFolderDependencies(dlFolder, includeTrashedEntries);
 
+		if (dlFolder.isApproved()) {
```
Comment (sergiogonzalez) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java]: We need to check that `dlFolder.getParentFolderId()` is not 0 because otherwise this will fail?

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java [hunk 3]
```diff
@@ -1130,7 +1145,7 @@ public void updateLastPostDate(long folderId, Date lastPostDate)
 			return;
 		}
 
-		dlFolder.setModifiedDate(dlFolder.getModifiedDate());
+		dlFolder.setModifiedDate(lastPostDate);
```
Comment (sergiogonzalez) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java]: why is this change needed? It's weird that a method name updateLastPostDate changes the modifieddate, isn't it?

Comment (renatodepaiva) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java]: While I agree that the method name is strange, if you take a look at the code was previously written, the modified date wasn't being modified at all in that method (it just take its value and applied it again).

I don't know the differences between the modified date and the last post date, but I can confirm that, regarding the folder, the date shown there is indeed the modified date.

Since even one of the arguments sent when the function is called is the modified date, in my opinion the correction is fine, since the intended date to be modified is indeed the modified date, not the last post date.

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java (hunk 1 above)
Comment (renatodepaiva) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java]: You're right, I will add the other condition to the if statement.

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java (hunk 2 above)
Comment (renatodepaiva) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java]: You're right, I will add the other condition to the if statement.

**Comment by renatodepaiva** [comment]
I will create the test case in the DLFileEntryLocalServiceTest, make the suggested corrections and re-send the pull request. Thanks!
